docs/api/browser-window.md

```diff
@@ -116,302 +84,46 @@ The [Page Visibility API][page-visibility-api] works as follows:
-      is not able to create a `<webview>` tag with a possibly malicious `preload`
-      script. You can use the `will-attach-webview` event on [webContents](web-contents.md)
-      to strip away the `preload` script and to validate or alter the
-      `<webview>`'s initial settings.
-    * `additionalArguments` String[] (optional) - A list of strings that will be appended
-      to `process.argv` in the renderer process of this app.  Useful for passing small
-      bits of data down to renderer process preload scripts.
-    * `safeDialogs` Boolean (optional) - Whether to enable browser style
-      consecutive dialog protection. Default is `false`.
-    * `safeDialogsMessage` String (optional) - The message to display when
-      consecutive dialog protection is triggered. If not defined the default
-      message would be used, note that currently the default message is in
-      English and not localized.
-    * `disableDialogs` Boolean (optional) - Whether to disable dialogs
-      completely. Overrides `safeDialogs`. Default is `false`.
-    * `navigateOnDragDrop` Boolean (optional) - Whether dragging and dropping a
-      file or link onto the page causes a navigation. Default is `false`.
-    * `autoplayPolicy` String (optional) - Autoplay policy to apply to
-      content in the window, can be `no-user-gesture-required`,
-      `user-gesture-required`, `document-user-activation-required`. Defaults to
-      `no-user-gesture-required`.
-    * `disableHtmlFullscreenWindowResize` Boolean (optional) - Whether to
-      prevent the window from resizing when entering HTML Fullscreen. Default
-      is `false`.
-    * `accessibleTitle` String (optional) - An alternative title string provided only
-      to accessibility tools such as screen readers. This string is not directly
-      visible to users.
-    * `spellcheck` Boolean (optional) - Whether to enable the builtin spellchecker.
-      Default is `true`.
-
-When setting minimum or maximum window size with `minWidth`/`maxWidth`/
-`minHeight`/`maxHeight`, it only constrains the users. It won't prevent you from
-passing a size that does not follow size constraints to `setBounds`/`setSize` or
-to the constructor of `BrowserWindow`.
-
-The possible values and behaviors of the `type` option are platform dependent.
-Possible values are:
-
-* On Linux, possible types are `desktop`, `dock`, `toolbar`, `splash`,
-  `notification`.
-* On macOS, possible types are `desktop`, `textured`.
-  * The `textured` type adds metal gradient appearance
-    (`NSTexturedBackgroundWindowMask`).
-  * The `desktop` type places the window at the desktop background window level
-    (`kCGDesktopWindowLevel - 1`). Note that desktop window will not receive
-    focus, keyboard or mouse events, but you can use `globalShortcut` to receive
-    input sparingly.
-* On Windows, possible type is `toolbar`.
+* `options` [BrowserWindowOptions](structures/browser-window-options.md) (optional)
```

Comment:
I think this might be one of the more heavily-used parts of Electron's documentation (@electron/wg-ecosystem perhaps has some stats?), so moving it is a big change. Also, it moves the `WebPreferences` documentation behind two separate clicks (first through to `BrowserWindowOptions` and then through to `WebPreferences`), where before they were directly available here.

I'm not necessarily against this reorganization, I just wanted to point out that it's a fairly major docs change and have some discussion about it.

Reply:
> I think this might be one of the more heavily-used parts of Electron's documentation

Good point! Google Analytics says it's the 5th most viewed page in the past 90 days, and within `/docs`, it's the 3rd one beyond the root docs page and the first app tutorial.

Reply:
I moved `BrowserWindowOptions` to a separate place because it now inherits from the `TopLevelWindowOptions`.

The `WebPreferences` is moved to structures because the new `WebContentsView` class will take it as parameter.

I don't know what's the best way to handle this. Maybe we can add a button on docs website to fill the information? (Sorry for pushing away the problem 😄 )